src/shared-components/MarkdownFileRenderer.tsx

```diff
@@ -17,56 +17,39 @@ async function getMarkdownHTML(fileName: string) {
 interface MarkdownFileRendererProps {
   fileName: string;
 }
-interface MarkdownFileRendererState {
-  html?: string;
-}
-
-class MarkdownFileRenderer extends React.Component<
-  MarkdownFileRendererProps,
-  MarkdownFileRendererState
-> {
-  constructor(props: MarkdownFileRendererProps) {
-    super(props);
-
-    this.state = {
-      html: undefined
-    };
 
-    const { fileName } = props;
-    if (!fileName) {
-      throw new TypeError(
-        'fileName must be specified for MarkdownFileRenderer'
-      );
-    }
-    getMarkdownHTML(fileName).then(html => this.setState({ html }));
-  }
-
-  render() {
-    return (
-      <div className="MarkdownFileRenderer">
-        {this.state.html ? (
-          [
-            <a
-              key="markdown-edit-link"
-              className="MarkdownFileRenderer-edit"
-              href={`https://github.com/codefordenver/members/edit/master/public/markdown/${
-                this.props.fileName
-              }.md`}
-            >
-              suggest edit
-            </a>,
+const MarkdownFileRenderer: React.FC<MarkdownFileRendererProps> = ({
+  fileName
+}) => {
+  const [html, setHtml] = useState<string | void>(undefined);
 
-            <div
-              key="markdown-content"
-              dangerouslySetInnerHTML={{ __html: this.state.html }}
-            />
-          ]
-        ) : (
-          <LoadingIndicator />
-        )}
-      </div>
-    );
+  if (!fileName) {
+    throw new TypeError('fileName must be specified for MarkdownFileRenderer');
   }
-}
+  getMarkdownHTML(fileName).then(html => setHtml(html));
+
+  return (
+    <div className="MarkdownFileRenderer">
+      {html ? (
+        [
+          <a
+            key="markdown-edit-link"
+            className="MarkdownFileRenderer-edit"
+            href={`https://github.com/codefordenver/members/edit/master/public/markdown/${fileName}.md`}
```

Comment:
Not in scope for this PR, but should this url be hard coded? It seems that this limits this component to only `.md` files stored there.

If these files are part of the `members` source code, why not just pull them locally instead of remotely? This seems like an anti-pattern.

Reply:
We are getting them locally (see the function: `getMarkdownHTML`). This link just pulls up a github editor page for that file so it can streamline edits from folks that might not otherwise be comfortable editing text in the code.